sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -449,17 +449,13 @@ private[spark] object SQLConf {
     doc = "When true, we could use `datasource`.`path` as table in SQL query"
   )
 
-  val SPECIALIZE_SINGLE_DISTINCT_AGG_PLANNING =
-    booleanConf("spark.sql.specializeSingleDistinctAggPlanning",
-      defaultValue = Some(true),
+  val AGG_PLANNING_15 =
+    booleanConf("spark.sql.aggregationPlanning15",
```

Comment:
After the discussion with @nongli @yhuai , we agreed that the only reason we have this flag here is to address the concern about performance regression for single distinct on low cardinality aggregation since 1.5, also this is the way Oracle usually did.